solc/CommandLineInterface.cpp

```diff
@@ -1087,61 +1099,91 @@ bool CommandLineInterface::assemble(
 	return true;
 }
 
+bool CommandLineInterface::assembleFromAssemblyJson()
+{
+	solAssert(m_fileReader.sourceCodes().size() == 1, "");
+
+	Json::Value assemblyJson;
+	m_assembly = make_unique<evmasm::Assembly>();
+	return util::jsonParseStrict(m_fileReader.sourceCodes().begin()->second, assemblyJson) &&
+				m_assembly->loadFromAssemblyJSON(assemblyJson, *m_assembly);
+}
+
 void CommandLineInterface::outputCompilationResults()
 {
 	handleCombinedJSON();
 
 	// do we need AST output?
 	handleAst();
 
-	if (
-		!m_compiler->compilationSuccessful() &&
-		m_options.output.stopAfter == CompilerStack::State::CompilationSuccessful
-	)
+	if (m_assembly)
```

Comment:
Can `m_assembly` an `m_compiler` both be present at the same time?

Reply:
No, only `m_assembly` or `m_compiler` is possible at the same time. Change this to `else if`.